packages/react-magma-dom/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,313 @@
+      window.addEventListener('resize', handleResize);
+      handleResize();
+
+      if (hasBreakpoint && breakpointStyle === BreakPointStyle.summaryView) {
+        setBreakPointSummaryView(true);
+      } else if (
+        hasBreakpoint &&
+        breakpointStyle === BreakPointStyle.hideLabels
+      ) {
+        setBreakPointHideLabels(true);
+      } else if (
+        (beyondBreakpoint && breakpointStyle === BreakPointStyle.hideLabels) ||
+        (beyondBreakpoint && BreakPointStyle.summaryView)
+      ) {
+        setBreakPointSummaryView(false);
+        setBreakPointHideLabels(false);
+      }
+
+      return () => window.removeEventListener('resize', handleResize);
+    }, []);
+
+    // Sets aria-current based on the active step, else false.
+    const ariaCurrent = index => {
+      if (currentStep === index) {
+        return true;
+      }
+      return false;
+    };
+
+    // Step states
+    const steps = React.Children.toArray(children);
+
+    const Steps = React.Children.map(
+      children,
+      (
+        child: React.ReactElement<React.PropsWithChildren<StepProps>>,
+        index
+      ) => {
+        if (child.type === Step) {
+          const stepStatusStyles =
+            currentStep >= index + 1
+              ? StepStatus.complete
+              : currentStep >= index
+              ? StepStatus.active
+              : StepStatus.incomplete;
+
+          const item = React.cloneElement(child, {
+            'aria-current': ariaCurrent(index),
+            children: currentStep === index ? child : null,
+            key: index,
+            isInverse: isInverse,
+            hideLabels: breakPointHideLabels ? true : hideLabels,
+            summaryView: summaryView || breakPointSummaryView,
+            stepStatus: stepStatusStyles,
+          });
+
+          return (
+            <>
+              {item}
+              {index !== steps.length - 1 && (
+                <StyledSeparator
+                  isInverse={isInverse}
+                  stepStatus={stepStatusStyles}
+                  theme={theme}
+                />
+              )}
+            </>
+          );
+        }
+      }
+    );
+
+    // When summaryView is set to true, this shows one step label and description at a time based on the active step below the Stepper component.
+
+    const StepLabels = React.Children.map(
+      children,
+      (
+        child: React.ReactElement<React.PropsWithChildren<StepProps>>,
+        index
+      ) => {
+        const item = child as React.ReactElement<
+          React.PropsWithChildren<StepProps>
+        >;
+
+        if (
+          item.type === Step &&
+          currentStep >= index &&
+          currentStep <= index
+        ) {
+          return child;
+        } else if (
+          // Retains last Step label and description
+          item.type === Step &&
+          currentStep >= +Steps.length &&
+          currentStep <= index + 1
+        ) {
+          return child;
+        }
```

Comment:
Since you return the same thing in both case, this can be combined to one statement
```
if (
    item.type === Step &&
    (currentStep >= index && currentStep <= index ||
    currentStep >= +Steps.length && currentStep <= index + 1)
  ) {
    return child;
  }
  ```